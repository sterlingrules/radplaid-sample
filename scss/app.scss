//
// Core
//
@import 'core/media';
@import 'core/constants';
@import 'core/columns';
@import 'core/colors';
@import 'core/typography';
@import 'core/images';
@import 'core/shadows';
@import 'core/themes';

//
// Lib
//
// @import 'react-select/scss/default';
@import 'react-dates/lib/css/datepicker';
// @import 'mapbox-gl/dist/mapbox-gl';
@import 'medium-editor/dist/css/medium-editor';
@import 'medium-editor/dist/css/themes/beagle';
@import '@glidejs/glide/dist/css/glide.core.min';

//
// Base
//
@import 'base/reset';
@import 'base/columns';
@import 'base/common';
@import 'base/general';
@import 'base/colors';
@import 'base/icons';
@import 'base/typography';
@import 'base/animations';
@import 'base/z-index';

//
// Modules
//
@import 'modules/progress';
@import 'modules/alerts';
@import 'modules/header/index';
@import 'modules/forms/index';
@import 'modules/buttons';
@import 'modules/states';
@import 'modules/violator';
@import 'modules/slider';
@import 'modules/list';
@import 'modules/notification';
@import 'modules/bubble';
@import 'modules/hero';
@import 'modules/modals';
@import 'modules/tooltips';
@import 'modules/calendar';
@import 'modules/accordion';
@import 'modules/drift';
@import 'modules/footer';
@import 'modules/error-boundary';
@import 'modules/more-menu';

//
// Features
//
@import 'home/index';
@import 'shows/index';
@import 'player/index';
@import 'profile/index';
@import 'settings/index';
