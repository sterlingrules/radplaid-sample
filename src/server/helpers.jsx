import keys from 'lodash/keys'
import isEmpty from 'lodash/isEmpty'
import intersection from 'lodash/intersection'
import { SEARCH_PARAMS } from './../app/constants.jsx'

/**
 * Removes br and p tags from strings
 *
 * @param str {String} - string to remove html characters from
 * @return {String} - cleaned string
 */
export const stripHtml = (str) => {
	if (typeof str !== 'string') {
		return str
	}

	return str
		.replace(/<(br|\/p).*?>/gm, ' ')
		.replace(/<(?:.|\n)*?>|"/gm, ' ')
		.trim()
}

/**
 * Formats `searchPath` from a query object
 *
 * @param {Object} query - key value query pairs
 * @return {String} - `searchPath`
 */
export const formatQueryObject = (query) => {
	let _keys = keys(query)
	let searchPath = ''

	_keys.forEach((key, index) => {
		if (index === 0) {
			searchPath += '?'
		}
		else {
			searchPath += '&'
		}

		searchPath += `${key}=${query[key]}`
	})

	return searchPath
}

/**
 * Determines whether or not a URL path is a search
 *
 * @param query {string} - URL query string
 * @return {Boolean} - Whether or not the query is a valid search
 */
export const isSearch = (query) => {
	let queryKeys = keys(query)

	return (
		!isEmpty(query) &&
		intersection(queryKeys, SEARCH_PARAMS).length > 0
	)
}

/**
 * Format meta title based on search query object
 *
 * @param {Object} - URL query string
 *		- @param query {String} - search query
 *		- @param cost {String} - cost (ie. free, under-15, over-15)
 *		- @param tag {String} - show tag
 *		- @param location {String} - city, state or zip
 * @return {String} - meta title for use in the HTML header
 */
export const formatQueryTitle = ({ query, cost, tag, location }) => {
	let reason = []
	let isCostCopy = ''

	// Handle cost title
	if (cost === 'free') {
		isCostCopy = 'Free '
	}
	else if (cost === 'paid') {
		isCostCopy = 'Paid '
	}

	// Handle query titles
	if (location) {
		reason.push(`near ${location}`)
	}

	if (query) {
		reason.push(`matching '${query}'`)
	}

	if (tag) {
		reason.push(`tagged '${tag}'`)
	}

	// Final format
	let heading = `All ${isCostCopy}Local Shows`

	if (query) {
		heading = 'Search Results'
	}

	return `Rad Plaid | ${heading} ${reason.length ? reason.join(' ').trim() : ''}`
}
