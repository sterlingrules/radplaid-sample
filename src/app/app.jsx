// import 'cross-fetch/polyfill'
import 'react-app-polyfill/stable'

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { isMobile, isAndroid } from './helpers/device-detect.js'
import { connect } from 'react-redux'
import { LastLocationProvider } from 'react-router-last-location'
import { withRouter, BrowserRouter as Router, Route, Switch, Link, Redirect } from 'react-router-dom'
import PrivateRoute from './authentication/components/private-route.jsx'
import URI from 'urijs'
import Helmet from 'react-helmet'
import moment from 'moment'

import { DATE_PATHS, META_TITLE, META_DESCRIPTION } from './constants.jsx'
import { delay, getQuery, events } from './helpers'
import { setUser, gTagReportConversion } from './helpers/analytics.jsx'
import Header from './common/header/index.jsx'
import ExitIntent from './common/exit-intent.jsx'
import Footer from './common/footer.jsx'
import Modals from './modals/index.jsx'
import Player from './player/index.jsx'
import * as AppActions from './redux/actions/app.jsx'
import User from './authentication/models/user.jsx'
import UserActions from './redux/actions/user.jsx'

// Views
import routes from './routes.jsx'
import NotFound from './common/404.jsx'

class App extends Component {
	static contextTypes = {
		store: PropTypes.object.isRequired
	}

	constructor(props) {
		super(props)

		this.state = {
			user: User.get()
		}
	}

	componentWillMount() {
		let metaToRemove = (typeof document !== 'undefined') ? document.querySelectorAll('[data-initial]') : []

		//
		// Handle MetaData setup. But we can probably
		// remove this once SSR is working properly
		if (!metaToRemove.length) {
			return
		}

		for (let i = 0; i < metaToRemove.length; i++) {
			metaToRemove[i].remove()
		}
	}

	componentDidMount() {
		if (typeof window === 'undefined') {
			return
		}

		let { isBrowser } = this.props
		let jwt = getQuery('jwt')
		let _jwt = localStorage.getItem('jwt')
		let token = getQuery('token')
		let uid = getQuery('uid')

		// Facebook Stuff
		let code = getQuery('code')
		let state = getQuery('state')

		let user

		let uri = new URI().removeQuery([
			'jwt',
			'token',
			'uid',
			'code',
			'state'
		])

		this.props.history.replace(`${uri.pathname()}${uri.search()}`)
		this.props.setIsBrowser(isBrowser)

		//
		// Check if we have a user,
		// then start the app
		if (this.state.user) {
			this.props.saveUser(this.state.user)
			this.props.initializeApp()
		}
		// Spotify Login
		else if (jwt) {
			User.set(jwt)
				.then((user) => {
					this.props.saveUser(user)
					this.props.initializeApp()
				})
		}
		// Passwordless Login
		else if (token && uid) {
			this.props.verifyLoginToken(token, uid)
		}
		// No user
		else {
			setTimeout(() => {
				this.props.initializeApp()
				User.clear()
			}, 100)
		}

		this._setViewportName()
		this._handlePushNotifications()
		this._listen()
	}

	componentWillUnmount() {
		this._listen('remove')
	}

	componentWillReceiveProps(nextProps) {
		let { user, progress, isBrowser } = nextProps
		let { location } = this.props
		let query = new URI().setSearch(location.search).search(true)

		if (!user || JSON.stringify(user) === '{}') {
			return
		}

		setUser(user.id, {
			"username": user.username,
			"$distinct_id": user.id,
			"$first_name": user.firstName,
			"$last_name": user.lastName,
			"$created": user.createdAt,
			"$last_login": new Date(),
			"$email": user.email
		})

		// New User, let's prompt for deets
		if (!user.username && progress.length === 0) {
			if (query.modal !== 'editprofile') {
				this.props.history.push('?modal=editprofile')
				gTagReportConversion()
			}
		}

		if (isBrowser !== this.props.isBrowser) {
			this.props.setIsBrowser(isBrowser)
		}
	}

	_onResize = (evt) => {
		delay(this._setViewportName, 1)
		events.publish('resize')

		if (isAndroid && typeof document !== 'undefined') {
			const isInput = (document.activeElement.tagName === 'INPUT' || document.activeElement.tagName === 'TEXTAREA')

			if (isInput) {
				setTimeout(() => {
					document.activeElement.scrollIntoView({
						behavior: 'smooth',
						block: 'end',
						inline: 'nearest'
					})
				}, 0)
			}
		}
	}

	_handlePushNotifications = () => {
		const { user } = this.props

		if (typeof OneSignal === 'undefined' || !user) {
			return
		}

		OneSignal.push(() => {
			OneSignal.setExternalUserId(user.id)
		})
	}

	_setViewportName = () => {
		let { setViewportName, setWindowProps } = this.props
		let viewportWidth = window.innerWidth
		let viewportHeight = window.innerHeight
		let htmlClassNames = document.documentElement.classList
		let viewportName = 'large'

		if (viewportWidth >= 768 && viewportWidth <= 1023) {
			viewportName = 'medium'
		}
		else if (viewportWidth <= 767) {
			viewportName = 'small'
		}

		setViewportName(viewportName)
		setWindowProps({
			width: viewportWidth,
			height: viewportHeight
		})
	}

	_listen = (state = 'add') => {
		const { setWindowLoad } = this.props

		window[`${state}EventListener`]('scroll', () => {
			if (isMobile) {
				return
			}

			events.publish('scroll', {
				top: window.scrollY,
				type: 'scroll'
			})
		})

		window[`${state}EventListener`]('touchmove', () => {
			events.publish('scroll', {
				top: window.scrollY,
				type: 'touchmove'
			})
		})

		window[`${state}EventListener`]('resize', this._onResize, false)
		window[`${state}EventListener`]('load', setWindowLoad, false)
	}

	render() {
		const {
			location,
			user,
			initialized,
			searchQuery,
			searchCost,
			searchTag,
			searchLocation,
			isSearching
		} = this.props

		let isNotice = false
		let isDatePath = (DATE_PATHS.indexOf(location.pathname) >= 0)
		let hasSearchParams = (searchQuery || searchCost || searchTag || searchLocation || isDatePath)
		let { steps } = this.state

		return (initialized && (
			<div className={`view is-logged-${user ? 'in' : 'out'} ${isNotice ? 'is-notice' : ''} ${isSearching ? 'is-searching' : ''} view-${location.pathname.substring(1).toLowerCase().replace(/\//g, '-') || 'home'}`}>
				<Helmet>
					<meta name="description" content={META_DESCRIPTION} />

					<meta property="fb:app_id" content={process.env.FACEBOOK_APP_ID} />

					<meta property="og:url" content={process.env.BASE_CLIENT_URL} />
					<meta property="og:type" content="website" />
					<meta property="og:title" content={META_TITLE} />
					<meta property="og:description" content={META_DESCRIPTION} />
					<meta property="og:image" content={`${process.env.BASE_CLIENT_URL}${process.env.ASSET_URL}/img/og.jpg`} />
					<meta property="og:locale" content="en_US" />

					<title>{META_TITLE}</title>

					<meta name="twitter:card" content="summary" />
					<meta name="twitter:site" content="@getradplaid" />
					<meta name="twitter:creator" content="@getradplaid" />
					<meta name="twitter:url" content={process.env.BASE_CLIENT_URL} />
					<meta name="twitter:title" content={META_TITLE} />
					<meta name="twitter:description" content={META_DESCRIPTION} />
					<meta name="twitter:image" content={`${process.env.BASE_CLIENT_URL}${process.env.ASSET_URL}/img/og.jpg`} />

					<link rel="canonical" href={process.env.BASE_CLIENT_URL} />
				</Helmet>

				{/*<ExitIntent user={user} />*/}

				<Header
					isNotice={isNotice}
					location={location} />

				<main className="main">
					<LastLocationProvider>
						<Switch>
							{routes.map((route) => {
								if (route.private) {
									return <PrivateRoute key={route.path} {...route} />
								}

								if (route.redirect) {
									return <Redirect key={route.path} to={route.redirect} exact={true} />
								}

								return <Route key={route.path} {...route} />
							})}
							<Route component={NotFound} />
						</Switch>
					</LastLocationProvider>
				</main>

				<Footer />

				<Player {...this.props} />
				<Modals {...this.props} />
			</div>
		))
	}
}

const mapStateToProps = ({ user, app }) => ({
	user: user.user,
	progress: app.progress,
	initialized: app.initialized,

	searchQuery: app.searchQuery,
	searchCost: app.searchCost,
	searchTag: app.searchTag,
	searchLocation: app.searchLocation,
	isSearching: app.isSearching
})

const mapDispatchToProps = dispatch => {
	return {
		initializeApp: () => {
			dispatch(AppActions.initializeSession())
		},

		loggingInUser: (state) => {
			dispatch(AppActions.loggingInUser(state))
		},

		setIsBrowser: (state) => {
			dispatch(AppActions.isBrowser(state))
		},

		logout: () => {
			dispatch(UserActions.logoutUser())
		},

		saveUser: (user) => {
			dispatch(UserActions.loginUser(user))
		},

		logoutUser: () => {
			dispatch(UserActions.logoutUser())
		},

		setViewportName: (name) => {
			dispatch(AppActions.setViewportName(name))
		},

		setWindowProps: (props) => {
			dispatch(AppActions.setWindowProps(props))
		},

		setWindowLoad: () => {
			dispatch(AppActions.setWindowLoad())
		},

		verifyLoginToken: (token, uid) => {
			dispatch(UserActions.verifyLoginToken(token, uid))
		},
	}
}

export default withRouter(
	connect(mapStateToProps, mapDispatchToProps)(App)
)
