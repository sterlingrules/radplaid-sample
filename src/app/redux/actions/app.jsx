import URI from 'urijs'
import * as TYPES from './../types.jsx'
import union from 'lodash/union'
import defaults from 'lodash/defaults'
import without from 'lodash/without'
import clone from 'lodash/clone'
import moment from 'moment'
import { isMobile } from './../../helpers/device-detect'
import User from './../../authentication/models/user.jsx'
import { request, requestSimple, requestPublic, requestOnce } from './../../helpers/request.jsx'
import { existsWithKey, getDateRange, getCurrentDate, localStorageExpire, replaceQuery } from './../../helpers'
import { DATE_PATHS, DEBUG, IS_APP } from './../../constants.jsx'
import { LOCATION_FALLBACK } from './../../constants.computed.jsx'
import ShowActions from './shows.jsx'

const updatePushNotificationLocation = ({ userId, name, coords = [] }) => {
	if (typeof window === 'undefined' || typeof OneSignal === 'undefined') {
		return
	}

	let data = {
		locationName: name,
		latitude: coords[0],
		longitude: coords[1],
	}

	if (userId) {
		data.userId = userId
	}

	OneSignal.push(() => {
		OneSignal.sendTags(data)
	})
}

export const initializeSession = (state = true) => {
	return (dispatch, getState) => {
		let { user, userLocation } = getState().app

		dispatch(loggingInUser(false))

		return dispatch({
			type: TYPES.INITIALIZE_SESSION,
			initialized: state
		})
	}
}

export const setConnection = (connection) => {
	return (dispatch, getState) => {
		const prevConnection = getState().app.connection

		if (connection !== prevConnection) {
			dispatch({
				type: TYPES.SET_CONNECTION,
				connection,
			})
		}

		if (!connection) {
			return setTimeout(() => {
				const callback = (err, reply) => {
					let isConnected = !!(reply && reply.status === 200)

					dispatch(setConnection(isConnected))

					if (isConnected) {
						dispatch(loadClear())
					}
				}

				requestSimple('/check-connection')
					.end(callback)
			}, 5000)
		}

		if (!prevConnection && connection) {
			dispatch(clearNotifications())
			dispatch(setNotification({
				status: 'success',
				message: 'Rad Plaid reconnected successfully. PleaseÂ refresh.',
				action: {
					label: 'Reload',
					icon: 'refresh',
					callback: () => window.location.reload()
				}
			}))
		}
	}
}

export const loggingInUser = (state = false) => {
	return {
		type: TYPES.LOGGING_IN,
		loggingIn: state
	}
}

export const startTutorial = () => {
	return {
		type: TYPES.RUN_TUTORIAL,
		runTutorial: true
	}
}

export const resetHome = () => {
	return (dispatch, getState) => {
		// let uri = new URI()

		let { searchSort, searchDateTo } = getState().app
		let { user } = getState().user
		let { location } = getState().routing
		let { shows, afterPage } = getState().shows
		let pathOptions = union(DATE_PATHS, [ '/' ])

		// Clears if we hit home while on home
		if (pathOptions.indexOf(location.pathname) >= 0 || !shows.length) {
			dispatch(resetSearch())
			dispatch(setSearchLiveStream(''))
			dispatch(ShowActions.apiFetchShowList({
				sort: User.getSort() || searchSort,
				cost: '',
				tag: '',
				venue: '',
				from: getCurrentDate(),
				to: searchDateTo || '',
				livestream: '',
				page: 0
			}, true))
		}
	}
}

export const setIsSearching = (isSearching) => {
	return (dispatch, getState) => {
		let uri = new URI()

		// Clear url params
		if (!isSearching) {
			uri.removeSearch('sort')
			uri.removeSearch('query')
			uri.removeSearch('location')
			uri.removeSearch('venue')
			uri.removeSearch('from')
			uri.removeSearch('to')
			uri.removeSearch('cost')
			uri.removeSearch('tag')
			uri.removeSearch('page')
		}

		dispatch({
			type: TYPES.SET_IS_SEARCHING,
			isSearching,
		})
	}
}

export const setSearchSort = (searchSort = 'all') => {
	return {
		type: TYPES.SET_SEARCH_SORT,
		searchSort,
	}
}

export const setSearchVenue = (searchVenue = '') => {
	return {
		type: TYPES.SET_SEARCH_VENUE,
		searchVenue,
	}
}

export const setSearchLiveStream = (searchLiveStream = '') => ({
	type: TYPES.SET_SEARCH_LIVE_STREAM,
	searchLiveStream,
})

export const setSearchQuery = (query) => {
	return {
		type: TYPES.SET_SEARCH_QUERY,
		searchQuery: query,
	}
}

export const setSearchLocation = (searchLocation) => {
	return {
		type: TYPES.SET_SEARCH_LOCATION,
		searchLocation,
	}
}

export const setSearchDateRange = (range = {}) => {
	let _range = defaults(range, { from: getCurrentDate(), to: '' })

	return {
		type: TYPES.SET_SEARCH_DATE_RANGE,
		from: _range.from,
		to: _range.to,
	}
}

export const setSearchCost = (cost = '') => {
	return {
		type: TYPES.SET_SEARCH_COST,
		cost,
	}
}

export const setSearchTag = (tag = '') => {
	return {
		type: TYPES.SET_SEARCH_TAG,
		tag,
	}
}

export const setInitialSearch = (query) => {
	return {
		type: TYPES.SET_INITIAL_SEARCH,
		...query,
	}
}

export const setUserIp = (ip) => {
	return {
		type: TYPES.SET_USER_IP,
		ip,
	}
}

export const resetSearch = () => {
	return {
		type: TYPES.RESET_SEARCH,
		resetSearch: new Date().getTime(),
	}
}


/**
 * Pushing an in-web-app notification
 *
 * @param messageType {String} - Notification type: success, error, info
 * @param message {String} - Notification body copy
 * @param messageTitle {String} - Notification title
 */
export const setNotification = (message = {}) => {
	return {
		type: TYPES.SET_NOTIFICATION,
		messageId: message.id || new Date().getTime(),
		messageType: message.status,
		messageTitle: message.title,
		message: message.message,
		messageAction: message.action,
		messageSticky: message.sticky,
		messageChildren: message.children
	}
}

export const clearNotifications = () => {
	return {
		type: TYPES.CLEAR_NOTIFICATION
	}
}

export const setHomeTutorial = (runHomeTutorial) => {
	return {
		type: TYPES.RUN_HOME_TUTORIAL,
		runHomeTutorial
	}
}

export const loadClear = () => {
	return {
		type: TYPES.LOAD_END,
		progressStatus: 'success',
		progress: [],
	}
}

export const loadStart = (name = 'app') => {
	return (dispatch, getState) => {
		let progress = clone(getState().app.progress) || []

		if (name) {
			if (progress.indexOf(name) >= 0) {
				return
			}

			progress.push(name)
		}

		return dispatch({
			type: TYPES.LOAD_START,
			progressStatus: '',
			progress
		})
	}
}

export const loadEnd = (err, name = 'app') => {
	return (dispatch, getState) => {
		let progress = clone(getState().app.progress) || []

		if (name && progress.indexOf(name) >= 0) {
			progress = without(progress, name)
		}
		// Clear all loaders
		else {
			progress = []
		}

		return dispatch({
			type: TYPES.LOAD_END,
			progressStatus: err ? 'error' : 'success',
			progress
		})
	}
}

export const setSession = (session = null) => {
	return {
		type: TYPES.SET_SESSION,
		session
	}
}

export const setViewportName = (viewportName = 'large') => {
	return {
		type: TYPES.SET_VIEWPORT_NAME,
		viewportName
	}
}

export const setWindowProps = (props) => {
	let defaultProps = {}

	if (typeof window !== 'undefined') {
		defaultProps.width = window.innerWidth
		defaultProps.height = window.innerHeight
	}

	props = defaults(props, defaultProps)

	return {
		type: TYPES.SET_WINDOW_PROPS,
		windowProps: props
	}
}

export const setWindowLoad = () => {
	return {
		type: TYPES.SET_WINDOW_LOAD,
		windowLoad: true
	}
}

export const setBlogPosts = (blogPosts) => {
	return {
		type: TYPES.SET_BLOG_POSTS,
		blogPosts
	}
}

export const isBrowser = (state = false) => {
	return {
		type: TYPES.IS_BROWSER,
		isBrowser: state
	}
}

/**
 * Claims
 */
export const setClaim = (type, data = {}) => {
	return {
		type: TYPES.SET_CLAIM,
		claim: {
			...data,
			type
		}
	}
}

export const clearClaim = () => {
	return {
		type: TYPES.SET_CLAIM,
		claim: {}
	}
}

/**
 * Set Report
 */
export const setReport = (type, data = {}) => {
	return {
		type: TYPES.SET_CLAIM,
		claim: {
			...data,
			type
		}
	}
}

/**
 * Search Action
 *
 * @param type {String} - The index to search. Default. `shows`
 * @param queryObject {Object} - An object with query params to search
 */
export const apiSearch = (type = 'shows', queryObject) => {
	return (dispatch, getState) => {
		let uri = queryObject ? new URI().setSearch(queryObject) : new URI()
		let { userLocation, ip, progress } = getState().app
		let { shows } = getState().shows
		let initialShowRequest = false
		let query = queryObject || uri.search(true)

		let settings = {
			path: `/search/${type}/${uri.search()}`
		}

		// Let's not start a new search
		// if one is in progress
		if (progress.indexOf('shows') >= 0) {
			return
		}

		if (process.browser) {
			dispatch(loadStart('shows'))
			dispatch(setIsSearching(true))
		}

		//
		// TODO: Clear `showlist` and display a loading message
		//
		if (!queryObject.page) {
			initialShowRequest = true
			dispatch(ShowActions.setShowTitle(''))
			dispatch(ShowActions.updateShowList({}))
		}

		requestPublic(settings)
			.end((err, reply) => {
				if (err) {
					dispatch(loadEnd(err, 'shows'))
					return
				}

				let { body } = reply
				let currentSlugs = initialShowRequest ? [] : shows.map(s => s.slug)

				body.shows = body.shows.filter(s => currentSlugs.indexOf(s.slug) < 0)

				if (query.page > 0) {
					dispatch(ShowActions.appendShowList(body))
				}
				else {
					replaceQuery({ sort: body.sort })

					dispatch(setSearchSort(body.sort))
					dispatch(ShowActions.setShowTitle(body.title))
					dispatch(ShowActions.updateShowList(body))

					if (Array.isArray(body.venues) && body.venues.length > 0) {
						dispatch({
							type: TYPES.SET_VENUES,
							venues: body.venues,
						})
					}
				}

				dispatch(loadEnd(err, 'shows'))
			})
	}
}

export const apiFetchBlogPosts = () => {
	return (dispatch, getState) => {
		requestPublic({ path: '/news' })
			.end((err, reply) => {
				if (err) {
					console.error(err)
				}

				let { body } = reply || {}

				dispatch(setBlogPosts(body))
			})
	}
}

export const apiFetchVenues = () => {
	return (dispatch, getState) => {
		let { venues, userLocation } = getState().app
		let coords = userLocation.coords ? userLocation.coords : []
		let settings = {
			path: `/venues/${coords.join(',')}`,
		}

		// We only need to query this once
		if (!venues || venues.length > 0 || !coords.length) {
			return
		}

		requestPublic(settings)
			.end((err, reply) => {
				if (err) {
					console.error(err)
				}

				if (!reply || !reply.body) {
					return
				}

				let { body } = reply || {}

				return dispatch({
					type: TYPES.SET_VENUES,
					venues: body.venues || []
				})
			})
	}
}

export const apiPostClaim = (claim) => {
	return (dispatch, getState) => {
		let settings = {
			method: 'post',
			path: '/claims',
			data: claim
		}

		request(settings)
			.end((err, reply) => {
				if (err) {
					console.error(err)
				}

				clearClaim()
				dispatch(clearNotifications())
				dispatch(setNotification({
					status: 'info',
					message: 'Claim received! You\'ll hear back from us within 24 hours.'
				}))
			})
	}
}

export const fetchScenes = () => {
	return (dispatch, getState) => {
		let { scenes } = getState().app
		let settings = {
			path: '/scenes'
		}

		if (scenes && scenes.length > 0) {
			return
		}

		requestPublic(settings)
			.end((err, reply) => {
				if (err) {
					console.error(err)
				}

				let { body } = reply

				return dispatch({
					type: TYPES.SET_SCENES,
					scenes: body
				})
			})
	}
}

export const setUserLocation = () => {
	return (dispatch, getState) => {
		if (IS_APP) {
			return
		}

		let { userLocation } = getState().app
		let { user } = getState().user
		let prevUserLocation = User.getLocation()

		// Check and set `prevUserLocation`
		// We want to use geolocation on mobile devices
		// if (existsWithKey(prevUserLocation, 'coords') && !isMobile) {
		if (prevUserLocation &&
			Array.isArray(prevUserLocation.coords) &&
			prevUserLocation.coords.length > 0) {

			// If we already have it in redux, stop
			if (userLocation.name == prevUserLocation.name) {
				return
			}

			return dispatch({
				type: TYPES.SET_USER_LOCATION,
				userLocation: prevUserLocation
			})
		}

		// If we didn't get a set location via IP
		// we'll need to ask the user for their location
		if (typeof navigator === 'undefined' || !('geolocation' in navigator)) {
			return
		}

		// Let's not nag for a users location
		// before they've logged in
		// if (!user) {
		// 	return
		// }

		dispatch(loadStart('userLocation'))

		navigator.geolocation.getCurrentPosition((position) => {
			let { latitude, longitude } = position.coords
			let coords = [ latitude, longitude ]
			let settings = {
				path: `/location/${latitude},${longitude}`
			}

			if (!latitude || !longitude) {
				return dispatch(loadEnd(null, 'userLocation'))
			}

			requestOnce(settings)
				.end((err, reply) => {
					dispatch(loadEnd(null, 'userLocation'))

					if (err || !reply) {
						return
					}

					let { body = {} } = reply

					body.coords = coords

					dispatch(clearNotifications())

					// if (location && location.coords.length > 0) {
					if (!prevUserLocation || prevUserLocation.name !== body.name) {
						User.setLocation(body)

						updatePushNotificationLocation({
							...body,
							userId: user ? user.id : null,
						})

						dispatch({
							type: TYPES.SET_USER_LOCATION,
							userLocation: body,
						})
					}
				})

		}, (error) => {
			if (DEBUG) {
				console.error('DEBUG: Error retrieving location ', error)
			}
		})
	}
}
