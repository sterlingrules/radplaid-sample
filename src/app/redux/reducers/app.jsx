import { DEFAULT_APP } from './../../constants.computed.jsx'
import User from './../../authentication/models/user.jsx'
import { getCurrentDate } from './../../helpers'
import * as TYPES from './../types.jsx'

const app = (previousState = DEFAULT_APP, action) => {
	let {
		from,
		activeShow,
		shows,
		addShow,
		user,
		claim,
		connection,
		runHomeTutorial,
		resetSearch,
		blogPosts,
		progressStatus,
		progressName,
		progress,
		isBrowser
	} = action

	switch(action.type) {
		case TYPES.INITIALIZE_SESSION:
			let { initialized } = action

			return {
				...previousState,
				initialized,
			}
		case TYPES.SET_CONNECTION:
			return {
				...previousState,
				connection,
			}
		case TYPES.LOGGING_IN:
			let { loggingIn } = action

			return {
				...previousState,
				loggingIn
			}
		case TYPES.IS_BROWSER:
			return {
				...previousState,
				isBrowser
			}
		case TYPES.SET_NOTIFICATION:
			let {
				messageId,
				messageSticky,
				messageTitle,
				messageType,
				messageChildren,
				message,
				messageAction
			} = action

			return {
				...previousState,
				messageClear: false,
				messageId,
				messageSticky,
				messageTitle,
				messageType,
				messageAction,
				messageChildren,
				message
			}
		case TYPES.CLEAR_NOTIFICATION:
			return {
				...previousState,
				messageClear: true,
				messageId: null
			}
		case TYPES.RUN_HOME_TUTORIAL:
			return {
				...previousState,
				runHomeTutorial
			}
		case TYPES.LOAD_START:
			return {
				...previousState,
				progressName,
				progressStatus,
				progress
			}
		case TYPES.LOAD_END:
			return {
				...previousState,
				progressName,
				progressStatus,
				progress
			}
		case TYPES.SET_SESSION:
			let { session } = action

			return {
				...previousState,
				session
			}
		case TYPES.SET_SCENES:
			let { scenes } = action

			return {
				...previousState,
				scenes
			}
		case TYPES.SET_VENUES:
			let { venues } = action

			return {
				...previousState,
				venues
			}
		case TYPES.SET_BLOG_POSTS:
			return {
				...previousState,
				blogPosts
			}
		case TYPES.SET_CLAIM:
			return {
				...previousState,
				claim
			}
		case TYPES.SET_VIEWPORT_NAME:
			let { viewportName } = action

			return {
				...previousState,
				viewportName
			}
		case TYPES.SET_WINDOW_LOAD:
			let { windowLoad } = action

			return {
				...previousState,
				windowLoad
			}
		case TYPES.SET_WINDOW_PROPS:
			let { windowProps } = action

			return {
				...previousState,
				windowProps
			}
		case TYPES.SET_IS_SEARCHING:
			let { isSearching } = action

			return  {
				...previousState,
				isSearching
			}
		case TYPES.SET_SEARCH_SORT:
			let { searchSort } = action

			return {
				...previousState,
				searchSort
			}
		case TYPES.SET_SEARCH_LIVE_STREAM:
			let { searchLiveStream } = action

			return {
				...previousState,
				searchLiveStream
			}
		case TYPES.SET_SEARCH_VENUE:
			let { searchVenue } = action

			return {
				...previousState,
				searchVenue
			}
		case TYPES.SET_SEARCH_QUERY:
			let { searchQuery } = action

			return {
				...previousState,
				searchQuery
			}
		case TYPES.SET_USER_IP:
			let { ip } = action

			return {
				...previousState,
				ip
			}
		case TYPES.SET_SEARCH_LOCATION:
			let { searchLocation } = action

			return {
				...previousState,
				searchLocation
			}
		case TYPES.SET_SEARCH_DATE_RANGE:
			let { to } = action

			return {
				...previousState,
				searchDateFrom: from,
				searchDateTo: to
			}
		case TYPES.SET_SEARCH_COST:
			let { cost } = action

			return {
				...previousState,
				searchCost: cost
			}
		case TYPES.SET_SEARCH_TAG:
			let { tag } = action

			return {
				...previousState,
				searchTag: tag
			}
		case TYPES.SET_INITIAL_SEARCH:
			return {
				...previousState,
				...action
			}
		case TYPES.SET_USER_LOCATION:
			let { userLocation } = action

			return {
				...previousState,
				userLocation
			}
		case TYPES.SET_WELCOME_TOTAL:
			let { welcomeTotal } = action

			return {
				...previousState,
				welcomeTotal
			}
		case TYPES.RESET_SEARCH:
			return {
				...previousState,
				searchSort: User.getSort(),
				searchQuery: '',
				searchLocation: '',
				searchCost: '',
				searchTag: '',
				searchDateFrom: getCurrentDate(),
				searchDateTo: '',
				searchVenue: '',
				resetSearch
			}
		case TYPES.RUN_TUTORIAL:
			return {
				...previousState,
				runTutorial
			}
		default:
			return previousState
	}
}

export default app
