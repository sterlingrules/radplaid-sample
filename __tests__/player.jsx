import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import store from './../src/app/redux/store.jsx'
import Player from './../src/app/player/index.jsx'
import { getCurrentDate } from './../src/app/helpers'

configure({ adapter: new Adapter() })

describe('Player', () => {
	xit('renders correctly', () => {
		expect(shallow(<Player {...{ store }} />)).toMatchSnapshot()
	})
})
