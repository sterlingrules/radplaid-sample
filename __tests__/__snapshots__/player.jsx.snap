// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders correctly 1`] = `
<Player
  currentTrack={null}
  isPlaying={false}
  loadStart={[Function]}
  onPause={[Function]}
  onPlay={[Function]}
  onPlaying={[Function]}
  pathname=""
  playNextTrack={[Function]}
  playlist={null}
  search=""
  searchCost=""
  searchDateFrom="2019-02-09T00:00:00-05:00"
  searchLocation=""
  searchQuery=""
  searchTag=""
  setNotification={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  trackPosition={0}
  user={null}
  viewportName="large"
/>
`;
